presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -89,15 +100,50 @@ private PinotExpression handleLogicalBinary(
         }
         List<RowExpression> arguments = call.getArguments();
         if (arguments.size() == 2) {
-            return derived(format(
-                    "(%s %s %s)",
-                    arguments.get(0).accept(this, context).getDefinition(),
-                    operator,
-                    arguments.get(1).accept(this, context).getDefinition()));
+            String def0 = arguments.get(0).accept(this, context).getDefinition();
+            String def1 = arguments.get(1).accept(this, context).getDefinition();
+            def1 = handleTimeStampAndDateCast(arguments.get(0), def1);
+            def0 = handleTimeStampAndDateCast(arguments.get(1), def0);
+            return derived(format("(%s %s %s)", def0, operator, def1));
         }
         throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("Unknown logical binary: '%s'", call));
     }
 
+    private String handleTimeStampAndDateCast(RowExpression rowExpression, String definition)
```

Comment:
By having handleTimeStampAndDateCast as written, its not clear what `definition` means.

Reply:
It's the value of timestamp or date(millsSinceEpoch/daysSinceEpoch)